api.go

```diff
@@ -82,7 +82,7 @@ func (d *Device) Query(request *http.Request) (*http.Response, error) {
 		createEnv := createResponse{}
 		createMultipartEnv := createMultipartResponse{}
 		queryPath := fmt.Sprintf(directModePath, url.PathEscape(d.ID()), "/query/object/multipart")
-		resp, err := d.tenant.regionalHttpClient.R().
+		resp, err := d.tenant.regionalHttpClients[d.Fqdn()].R().
```

Comment:
Not sure if this will happen. What if the d.Fqdn() is not part of the configured regions?
Because I think the get device info is a global api
If we can get devices from other regions, perhaps we need to filter during getDevices() and device activate and deactivate
When you can test multi-region, can you verify this?

Reply:
> Not sure if this will happen. What if the d.Fqdn() is not part of the configured regions? Because I think the get device info is a global api If we can get devices from other regions, perhaps we need to filter during getDevices() and device activate and deactivate When you can test multi-region, can you verify this?

Added checks to handle this and tested it.